UnicornEngine/video/source/vulkan/Memory.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#include <unicorn/video/vulkan/Memory.hpp>
+
+namespace unicorn
+{
+namespace video
+{
+namespace vulkan
+{
+Memory::Memory() : m_initialized(false)
+                 , m_memory(nullptr)
+{
+}
+
+Memory::~Memory()
+{
+    Free();
+}
+
+bool Memory::IsInitialized() const
+{
+    return m_initialized;
+}
+
+vk::Result Memory::Allocate(
+    vk::Device device,
+    uint32_t typeFilter,
+    vk::PhysicalDeviceMemoryProperties physMemProperties,
+    vk::MemoryPropertyFlagBits reqMemProperties,
+    uint64_t allocSize)
+{
+    m_device = device;
```

Comment:
Same as with `Image::Create` - please check your init flag to make sure you're not leaving the object's state broken and that you're doing a proper clean up when reinitializing the object